### PR TITLE
Fail Contract test if the required handlers are not present

### PR DESCRIPTION
**Pull request by anshikg**
*Issue #, if available:*

*Description of changes:* This CR ensures contract test fail if a resource doe snot have Create/Delete/Read handler as they are required handlers

*Test:* Ran contract test for a resource which does not have a read handler
```
========================================================================================================================================================================================================================================= test session starts =========================================================================================================================================================================================================================================
platform darwin -- Python 3.8.9, pytest-6.2.3, py-1.9.0, pluggy-0.13.1 -- /Users/anshikg/workspace/cloudformation-cli/env/bin/python3
cachedir: .pytest_cache
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm/.hypothesis/examples')
rootdir: /Users/anshikg/workspace/aws-cloudformation-resource-providers-cloudwatch/aws-cloudwatch-alarm, configfile: ../../../../../private/var/folders/zh/rhw_046j7bq4d88hjn671hqjy323j8/T/pytest_011vp1b7.ini
plugins: random-order-1.0.4, localserver-0.5.0, cov-2.10.0, hypothesis-5.19.3, black-0.3.10
collected 14 items / 5 deselected / 9 selected                                                                                                                                                                                                                                                                                                                                                                                                                                                        

handler_create.py::contract_create_delete FAILED                                                                                                                                                                                                                                                                                                                                                                                                                                                [ 11%]
handler_create.py::contract_create_duplicate ERROR                                                                                                                                                                                                                                                                                                                                                                                                                                              [ 22%]
handler_delete.py::contract_delete_list ERROR                                                                                                                                                                                                                                                                                                                                                                                                                                                   [ 33%]
handler_delete.py::contract_delete_update ERROR                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ 44%]
handler_delete.py::contract_delete_delete ERROR                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ 55%]
handler_delete.py::contract_delete_create ERROR                                                                                                                                                                                                                                                                                                                                                                                                                                                 [ 66%]
handler_misc.py::contract_check_asserts_work PASSED                                                                                                                                                                                                                                                                                                                                                                                                                                             [ 77%]
handler_update.py::contract_update_list_success ERROR                                                                                                                                                                                                                                                                                                                                                                                                                                           [ 88%]
handler_update_invalid.py::contract_update_non_existent_resource FAILED          
```



By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/rpdk/core/contract/resource_client.py [hunk 1]
```diff
@@ -587,6 +587,8 @@ def _call(self, payload):
     def call_and_assert(
         self, action, assert_status, current_model, previous_model=None, **kwargs
     ):
+        if not self.has_required_handlers_exist():
+            raise ValueError("Create/Read/Delete/List handlers are required")
```
Comment (priyap286) [inline src/rpdk/core/contract/resource_client.py:591]: The error message says List is required but the method does not enforce it?

Comment (anshikg) [reply src/rpdk/core/contract/resource_client.py:591]: My bad, I missed it. I was thinking of asserting list but decided against it

**Review by priyap286 — APPROVED** [review]

File: src/rpdk/core/contract/resource_client.py [hunk 2]
```diff
@@ -632,3 +634,12 @@ def call(self, action, current_model, previous_model=None, **kwargs):
 
     def has_update_handler(self):
         return "update" in self._schema["handlers"]
+
+    def has_required_handlers_exist(self):
```
Comment (gandhek) [inline src/rpdk/core/contract/resource_client.py:638]: Can we rename this to "has_required_handlers"? exist seems to be redundant

**Review by gandhek — CHANGES_REQUESTED** [review]

File: src/rpdk/core/contract/resource_client.py (hunk 2 above)
Comment (anshikg) [reply src/rpdk/core/contract/resource_client.py:638]: makes sense.

**Review by gandhek — APPROVED** [review]

**Comment by PatMyron** [comment]
Looks like all the current [public resource schemas](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html) comply 👍 
```python
print(sorted(resource_spec.get("handlers", []).keys()))

 234 ['create', 'delete', 'list', 'read', 'update']
  43 ['create', 'delete', 'read', 'update']
  31 ['create', 'delete', 'list', 'read']
  13 ['create', 'delete', 'read']
```

**Pull request merged**